mix.exs

```diff
@@ -100,6 +100,7 @@ defmodule Igniter.MixProject do
       {:spitfire, "~> 0.1 and >= 0.1.3"},
       {:sourceror, "~> 1.4"},
       {:jason, "~> 1.4"},
+      {:phx_new, "~> 1.7", runtime: false},
```

Comment:
Is this okay?

Reply:
🤔 This one is tough. I don't actually think we need this. IIRC `phx_new` will be available as long as the user has the archive installed? Which could mean that we can just at the start say "please install `phx_new` to use this task" if the module is not defined.

Reply:
So we could have it as a test only dependency in that case.

Reply:
> I don't actually think we need this. IIRC `phx_new` will be available as long as the user has the archive installed? Which could mean that we can just at the start say "please install `phx_new` to use this task" if the module is not defined.

Should we have it as an optional dependency, in that case? That seems like the best of both worlds: It doesn't become a transient dependency that all apps using Igniter are required to fetch, but it still allows Igniter to test against it and specify a version spec that Igniter's compatible with.

Reply:
🤔 I think that would make sense, yes. Since we'd be switching in our code on the module being compiled, it wouldn't matter if it's from a dep or the archive.

Reply:
Changed to `optional: true` and added a [check](https://github.com/ash-project/igniter/pull/140/files#diff-9b04be38bfa30b2c4a6e746774bb6a48a179fc844fcb5710477eae3801d1debfR110-R117). Tested in a project, it will ask to install `phx_new` if not available.